File: python/bulk_lookup.py
import os
import httpx

API_KEY = os.environ["NPI_API_KEY"]
BASE_URL = os.environ.get("NPI_API_BASE_URL", "https://healthproviderapi.com")
NPIS = ["1003000126", "1932100864", "1851789159"]


def bulk_lookup(npis: list[str]) -> dict:
    response = httpx.post(
        f"{BASE_URL}/api/v1/npi/bulk",
        json={"npis": npis},
        headers={"Authorization": f"Bearer {API_KEY}"},
        timeout=30.0,
    )
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    data = bulk_lookup(NPIS)
    counts = data["meta"]["counts"]
    print(f"Requested: {counts['requested']}, Found: {counts['found']}, Not found: {counts['notFound']}")
    print(data)

File: node/bulk-lookup.js
import 'dotenv/config';

const API_KEY = process.env.NPI_API_KEY;
const BASE_URL = process.env.NPI_API_BASE_URL || 'https://healthproviderapi.com';

const NPIS = ['1003000126', '1932100864', '1851789159'];

async function bulkLookup(npis) {
  const res = await fetch(`${BASE_URL}/api/v1/npi/bulk`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${API_KEY}`,
      'Content-Type': 'application/json',
      'Accept': 'application/json',
    },
    body: JSON.stringify({ npis }),
  });

  if (!res.ok) {
    const body = await res.json();
    throw new Error(`API error ${res.status}: ${body.error.code} — ${body.error.message}`);
  }

  const data = await res.json();
  const counts = data.meta.counts;
  console.log(`Requested: ${counts.requested}, Found: ${counts.found}, Not found: ${counts.notFound}`);
  console.log(JSON.stringify(data, null, 2));
  return data;
}

bulkLookup(NPIS).catch(console.error);
